### PR TITLE
special log for executing  & hot load code

### PR DESCRIPTION
**Pull request by refactor**
* console-log the executing src-code, with format "{module:fun#line}"
  * print the partition num as HEX string, and padding the prefix zeros
  * hot-load, when src-code modified

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -1,6 +1,9 @@
-{erl_opts, [debug_info, {parse_transform, lager_transform}]}.
+{erl_opts, [debug_info, {parse_transform, lager_transform},{lager_extra_sinks, [mylog]}]}.
 {deps, [{riak_core, "3.0.9", {pkg, riak_core_ng}},
-        {lager, "3.5.1"}]}.
+	{recon, "2.3.4"},
```
Comment (facundoolano) [inline rebar.config:3]: We prefer not to add extra dependencies (part of the problem with other tutorials is that they had many dependencies so there where lots of spots that failed when they became outdated).

**Comment by unbalancedparentheses** [comment]
It is only a tutorial. We don't want to add unnecessary deps.

**Comment by refactor** [comment]
for the tutorial, the hot-code reload is really convenient and time-saver, so we do not need to re-make and restart vm when modified code.

**Comment by unbalancedparentheses** [comment]
This should be used for that case instead: https://github.com/vans163/rebar3_auto

It needs to be configured/install on the user's rebar3 profile so no need to add anything to the repo. Thanks for contributing but we are not big fans on unnecessary dependencies.

**Comment by refactor** [comment]
for rebar3_auto, on Linux you need to install inotify-tools.

**Pull request closed without merging**